corehq/apps/locations/tests/test_bulk_management.py

```diff
@@ -221,46 +222,54 @@ class MockLocationStub(LocationStub):
     def lookup_old_collection_data(self, old_collection):
         try:
             return super(MockLocationStub, self).lookup_old_collection_data(old_collection)
-        except AttributeError:
+        except (AttributeError, KeyError):
             # This will happen if the LocationStub is not new and an old_collection is not given
             self.db_object = MagicMock()
 
 
+@attr.s
+class TestLocationCollection(LocationCollection):
+
+    domain_name = 'location-bulk-management'
+
+    types = attr.ib(default=attr.Factory(list))
+    locations = attr.ib(default=attr.Factory(list))
+
+    def custom_data_validator(self):
+        return lambda data: False
+
+
+class IngoreOldLocationTreeValidator(LocationTreeValidator):
+    """Skip checks that depend on self.old_collection"""
```

Comment:
> What does this mean conceptually? Skip things that rely on the db?

Basically yes. This allows moving some test logic out of `LocationTreeValidator`. For example:

https://github.com/dimagi/commcare-hq/blob/8272b2067782c34206a9d1a19078337f3d331ea2/corehq/apps/locations/bulk_management.py#L627-L630

Using a subclass this way allows us to move that test logic into the test code only.

> Does passing in `TestLocationCollection()` as the fallback if `old_collection` is missing not suffice to make these tests pass?

It's not quite that simple because the tests that require this were not written in a way that makes it easy to just pass in a `TestLocationCollection()`. It would have to be populated with the correct locations to have that work, but I didn't want to go to the bother of figuring out how to do that and then also verify that the tests were still testing the same thing they were before the change.